sqlparser/token.go

```diff
@@ -777,6 +777,13 @@ func (tkn *Tokenizer) scanLiteralIdentifier() (int, []byte) {
 	if buffer.Len() == 0 {
 		return LEX_ERROR, buffer.Bytes()
 	}
+
+	// Double-quoted identifiers in PostgreSQL are case-sensitive.
+	// We need to remember which identifiers were wrapped with quotes so the query
+	// can be properly recreated later from the AST using .Format() method.
+	if quoteSeen && tkn.IsPostgreSQL() {
+		return DOUBLE_QUOTE_STRING, buffer.Bytes()
```

Comment:
maybe will be better to return `stringTokenType[ch]`? instead of hardcoding type of token without relation to value that came

Reply:
Well, yes, but for PostgreSQL `tkn.dialect.QuoteHandler().IsIdentifierQuote(...)` only compares with `"`, so other type of quote can't appear here. To be able to use `stringTokenType[...]` I also need to track the quote somewhere, introduce another var and set it next to `quoteSeen = true`. You meant to make like this, right?

Also, it's kinda confusing for me that we enter this function (`scanLiteralIdentifier`) only if `tkn.dialect.QuoteHandler().IsIdentifierQuote(byte(ch))`, and the next chunk
```
if tkn.dialect.QuoteHandler().IsStringLiteralQuote(byte(ch)) {
	return tkn.scanString(ch, stringTokenType[ch])
}
```
is also about quotes. Then when the usual non-quote-wrapped identifier is processed?

Reply:
> so other type of quote can't appear here

so you can return using this map to support mysql too and avoid hardcoding

>  I also need to track the quote somewhere, introduce another var and set it next to

hm, actually we can aggregate it. store `var quote *byte` and instead of `false` set nil. And at the end if it's not nil, return `stringTokenType[*quote]`

> Also, it's kinda confusing for me that we enter this function (scanLiteralIdentifier) only if tkn.dialect.QuoteHandler().IsIdentifierQuote(byte(ch)), and the next chunk
> 
> > if tkn.dialect.QuoteHandler().IsStringLiteralQuote(byte(ch)) {
> > 	return tkn.scanString(ch, stringTokenType[ch])
> > }
> 
> is also about quotes. Then when the usual non-quote-wrapped identifier is processed?

Quoted values can be identifiers or string literals. It depends on RDBMS (pgsql or mysql) and settings (in case of mysql). So at first we check is it identifier. After that it actually can be only string literal.

Identifiers can be quote wrapped and not quote wrapped. Not wrapped returned as `ID` from our tokenizer and it depends on grammar context to decide is it table's or column's or else's identifier.